gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql

```diff
@@ -0,0 +1,111 @@
+/* gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql */
+
+/*
+ This is part of arenadata_toolkit API for ADB Bundle.
+ This function collects information in db_files_current and db_files_history tables.
+ */
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_collect_table_stats()
+RETURNS VOID
+AS $$
+BEGIN
+	IF NOT EXISTS (
+		SELECT
+		FROM pg_partition pp
+		JOIN pg_partition_rule pr ON pr.paroid = pp.oid
+		WHERE
+			pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
+			AND
+			CASE

```

Comment:
why still we need case? test passes and without it
```diff
diff --git a/gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql b/gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql
index ebb3ea672dc..82913ad0fb6 100644
--- a/gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql
+++ b/gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql
@@ -15,16 +15,13 @@ BEGIN
 		WHERE
 			pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
 			AND
-			CASE
-				WHEN pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
-				THEN now() BETWEEN
+				now() BETWEEN
 					CAST(substring(pg_get_expr(pr.parrangestart, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
 						AS TIMESTAMP WITHOUT TIME ZONE)
 					AND
 					CAST(substring(pg_get_expr(pr.parrangeend, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
 						AS TIMESTAMP WITHOUT TIME ZONE)
-				ELSE FALSE
-			END)
+			)
 	THEN
 		EXECUTE FORMAT($fmt$ALTER TABLE arenadata_toolkit.db_files_history SPLIT DEFAULT PARTITION
 			START (date %1$L) INCLUSIVE
```

Reply:
> why still we need case? test passes and without it

I suppose, your variant of the query is planned and executed in a different way when it is inside the function during the test.
I've checked the query in isolation and it fails - refer to log below.
And the plan for the query is not safe - it will apply filtering by partition borders without pre-filtering by the table name.

```
postgres=# -- check that adb_collect_table_stats is executed without an error
select arenadata_toolkit.adb_collect_table_stats();
 adb_collect_table_stats 
-------------------------
 
(1 row)

postgres=# 
postgres=# SELECT
                FROM pg_partition pp
                JOIN pg_partition_rule pr ON pr.paroid = pp.oid
                WHERE
                        pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
                        AND
                        now() BETWEEN
                                CAST(substring(pg_get_expr(pr.parrangestart, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
                                        AS TIMESTAMP WITHOUT TIME ZONE)
                                AND
                                CAST(substring(pg_get_expr(pr.parrangeend, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
                                        AS TIMESTAMP WITHOUT TIME ZONE);
ERROR:  invalid input syntax for type timestamp: "'10'"
postgres=# 
postgres=# explain SELECT
                FROM pg_partition pp
                JOIN pg_partition_rule pr ON pr.paroid = pp.oid
                WHERE
                        pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
                        AND
                        now() BETWEEN
                                CAST(substring(pg_get_expr(pr.parrangestart, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
                                        AS TIMESTAMP WITHOUT TIME ZONE)
                                AND
                                CAST(substring(pg_get_expr(pr.parrangeend, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
                                        AS TIMESTAMP WITHOUT TIME ZONE);
                                                                                                                                                                                       QUERY PLAN                  
                                                                                                                                                                      
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
----------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Hash Join  (cost=1.01..3.05 rows=4 width=0)
   Hash Cond: (pr.paroid = pp.oid)
   ->  Seq Scan on pg_partition_rule pr  (cost=0.00..2.00 rows=1 width=4)
         Filter: (('2024-04-22 09:15:15.521227+10'::timestamp with time zone >= ("substring"(pg_get_expr(parrangestart, parchildrelid), '#"%#"::%'::text, '#'::text))::timestamp without time zone) AND ('2024-04-2
2 09:15:15.521227+10'::timestamp with time zone <= ("substring"(pg_get_expr(parrangeend, parchildrelid), '#"%#"::%'::text, '#'::text))::timestamp without time zone))
   ->  Hash  (cost=1.00..1.00 rows=1 width=4)
         ->  Seq Scan on pg_partition pp  (cost=0.00..1.00 rows=1 width=4)
               Filter: (parrelid = '16413'::oid)
 Optimizer: Postgres query optimizer
(8 rows)

postgres=# 
```

And in case the tables with partitioning by bigint are created at the beginning of the test, the test doesn't pass with your variant of the query.
So, I've updated the test. And better to keep current solution with 'case' as it should guarantee that partition borders are checked only for our target table.